hepdata/modules/records/utils/records_update_utils.py

```diff
@@ -0,0 +1,96 @@
+import datetime
+import requests
+import math
+
+from hepdata.modules.records.utils.doi_minter import generate_dois_for_submission
+from hepdata.modules.submission.api import get_latest_hepsubmission
+from hepdata.modules.records.utils.workflow import update_record
+from hepdata.modules.inspire_api.views import get_inspire_record_information
+from hepdata.ext.elasticsearch.api import index_record_ids
+from hepdata.modules.email.api import notify_publication_update
+
+
+def update_record_info(inspire_id, recid='', send_email=False):
+    inspire_id = inspire_id.replace("ins", "")
+
+    if recid == '':
+        hep_submission = get_latest_hepsubmission(inspire_id=inspire_id)
+        if hep_submission is None:
+            print("Failed to retrieve hep submission for {0}".format(inspire_id))
+            return
+        recid = hep_submission.publication_recid
+
+    updated_record_information, status = get_inspire_record_information(inspire_id)
+
+    if status == 'success':
+        record_information = update_record(recid, updated_record_information)
+    else:
+        print("Failed to retrieve publication information for {0}".format(inspire_id))
+        return
+
+    if recid == '' and hep_submission.overall_status == 'finished':
+        index_record_ids([record_information["recid"]])
+        generate_dois_for_submission.delay(inspire_id=inspire_id)  # update metadata stored in DataCite
+        if send_email:
+            notify_publication_update(hep_submission, record_information)   # send email to all participants
+
+
+def get_all_updated_records_since_date(date, verbose=False):
+    """
+    Returns all inspire ids of records updated since the specified date (YYYY-MM-DD).
+    Alternatively, date can be a number, denoting days since today (-1 = yesterday).
+    Only records updated on that specific date will then be returned.
+    """
+
+    date_is_delta_from_today = (type(date) is int or type(date) is str and date.isdigit())
+
+    if verbose:
+        if date_is_delta_from_today:
+            print("Obtaining inspire ids of records updated {} days ago.".format(abs(int(date))))
+        else:
+            print("Obtaining inspire ids of records updated since {}.".format(date))
+
+    if date_is_delta_from_today:
+        specified_time = datetime.datetime.today() + datetime.timedelta(days=-int(date))
+    else:
+        specified_time = datetime.datetime.strptime(date, "%Y-%m-%d")
+
+    page = 1
+    counter = 0
+    ids = []
+
+    while page < 1000:
+
+        if date_is_delta_from_today:
+            response = requests.get('https://inspirehep.net/api/literature?sort=mostrecent&size=25&page={}'.format(page) +
+                                    '&q=external_system_identifiers.schema%3AHEPData%20and%20legacy_version%3A{}%2A'.format(specified_time.strftime("%Y%m%d")))
+        else:
+            response = requests.get("https://inspirehep.net/api/literature?q=external_system_identifiers.schema:HEPData&page={}".format(page))
```

Comment:
This seems inefficient to get all records from INSPIRE and later discard some of them.  I think a search like `legacy_version:20200629*` can be used for **on** a certain day and `legacy_version:20200629+` for **since** a certain day.  Can you please use your `resilient_requests` function instead of the normal `requests` which is failing for me if I try `get_all_updated_records_since_date('2020-06-29')?

Reply:
I feel like there is an issue with the server api for records updated recently, for instance this [Records updated on 2020-07-15](https://inspirehep.net/literature?sort=mostrecent&size=10&page=1&q=external_system_identifiers.schema%3AHEPData%20and%20legacy_version%3A20200715%2A) works, but if I add ```api/``` befrore ```literature``` I constantly get a ```502 Bad Gateway``` error. If I try a older date, say [Records updated on 2020-07-13](https://inspirehep.net/literature?sort=mostrecent&size=10&page=1&q=external_system_identifiers.schema%3AHEPData%20and%20legacy_version%3A20200713%2A) then I can add ```api/``` without getting ```502 Bad Gateway```. I'm not sure what to do about it. For the code I've switched to ```resilient_requests``` but it simply exceeds the max retries number.

Reply:
Implemented both usage of ```resilient_requests``` and two functions for ```on``` date and ```since``` date, by using ```*``` or ```+``` in the ```legacy_version``` query. The two functions call a 'private' (i.e. starting with ```_```) function with an ```on_or_since``` argument switch, in order to reduce code duplication. However, I think it is more user friendly to have two separate functions instead of an argument switch, hence the code structure.